### ISSUE TITLE
Fresh installation of kube-prometheus 8.22.0 + bitnami/thanos 12.13.13 yields minio permission denied error 

### ISSUE THREAD

**Issue by lony** [bitnami/charts#20598]
### Name and Version

bitnami/thanos 12.13.13

### What architecture are you using?

amd64

#### I Setup bitnami/kube-prometheus 8.22.0

1. Create new folder and add Chart.yaml

```yaml
apiVersion: v2
name: thanos-chart
version: 1.0.0

dependencies:
  - name: kube-prometheus
    version: 8.22.0
    repository: https://charts.bitnami.com/bitnami
```
2. Add values.yaml file

```yaml
kube-prometheus:
  prometheus:
    thanos:
      create: true
```
3. Create namespace thanos
4. Run `helm install --namespace thanos thanos-inst .`

#### II Setup bitnami/thanos 12.13.13 (inside same meta chart)

1. Create a new folder and add Chart.yaml

```yaml
apiVersion: v2
name: thanos-chart
version: 1.0.0

dependencies:
  - name: kube-prometheus
    version: 8.22.0
    repository: https://charts.bitnami.com/bitnami
  - name: thanos
    version: 12.13.13
    repository: https://charts.bitnami.com/bitnami
```
2. Add values.yaml file

```yaml
thanos:
  objstoreConfig: |-
    type: s3
    config:
      bucket: thanos
      endpoint: {{ include "thanos.minio.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local:9000
      access_key: minio
      secret_key: minio123
      insecure: true
  query:
    dnsDiscovery:
      sidecarsService: kube-prometheus-prometheus-thanos
      sidecarsNamespace: thanos
  bucketweb:
    enabled: true
  compactor:
    enabled: true
  storegateway:
    enabled: true
  ruler:
    enabled: true
    alertmanagers:
      - http://kube-prometheus-alertmanager.thanos.svc.cluster.local:9093
    config: |-
      groups:
        - name: "metamonitoring"
          rules:
            - alert: "PrometheusDown"
              expr: absent(up{prometheus="thanos/kube-prometheus"})
  metrics:
    enabled: true
    serviceMonitor:
      enabled: true
  minio:
    enabled: true
    auth:
      rootPassword: minio123
      rootUser: minio
    monitoringBuckets: thanos
    accessKey:
      password: minio
    secretKey:
      password: minio123
  volumePermissions:
    enabled: true
```
3. Create namespace thanos
4. Run `helm install --namespace thanos thanos-inst .`

### Are you using any custom parameters or values?

```yaml
kube-prometheus:
  prometheus:
    thanos:
      create: true

thanos:
  objstoreConfig: |-
    type: s3
    config:
      bucket: thanos
      endpoint: {{ include "thanos.minio.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local:9000
      access_key: minio
      secret_key: minio123
      insecure: true
  query:
    dnsDiscovery:
      sidecarsService: kube-prometheus-prometheus-thanos
      sidecarsNamespace: thanos
  bucketweb:
    enabled: true
  compactor:
    enabled: true
  storegateway:
    enabled: true
  ruler:
    enabled: true
    alertmanagers:
      - http://kube-prometheus-alertmanager.thanos.svc.cluster.local:9093
    config: |-
      groups:
        - name: "metamonitoring"
          rules:
            - alert: "PrometheusDown"
              expr: absent(up{prometheus="thanos/kube-prometheus"})
  metrics:
    enabled: true
    serviceMonitor:
      enabled: true
  minio:
    enabled: true
    auth:
      rootPassword: minio123
      rootUser: minio
    monitoringBuckets: thanos
    accessKey:
      password: minio
    secretKey:
      password: minio123
  volumePermissions:
    enabled: true
```

### What is the expected behavior?

Everything is running and has the Running status

### What do you see instead?

minio and storegateway pod are stuck in CrashLoopBackOff state

minio log

```
 20:53:03.97 
 20:53:03.97 Welcome to the Bitnami minio container
 20:53:03.97 Subscribe to project updates by watching https://github.com/bitnami/containers
 20:53:03.98 Submit issues and feature requests at https://github.com/bitnami/containers/issues
 20:53:03.98 
 20:53:03.98 INFO  ==> ** Starting MinIO setup **
/opt/bitnami/scripts/libminio.sh: line 364: /bitnami/minio/data/.root_user: Permission denied
Stream closed EOF for thanos/thanos-inst-minio-59785f96cd-4p8nv (minio)
```              

storegateway log

```
ts=2023-11-02T20:52:36.99920273Z caller=factory.go:53 level=info msg="loading bucket configuration"
ts=2023-11-02T20:52:36.99961213Z caller=inmemory.go:179 level=info msg="created in-memory index cache" maxItemSizeBytes=131072000 maxSizeBytes=262144000 maxItems=maxInt
ts=2023-11-02T20:52:36.999994116Z caller=options.go:26 level=info protocol=gRPC msg="disabled TLS, key and cert must be set to enable"
ts=2023-11-02T20:52:37.001122643Z caller=store.go:520 level=info msg="starting store node"
ts=2023-11-02T20:52:37.001223879Z caller=store.go:418 level=info msg="initializing bucket store"
ts=2023-11-02T20:52:37.001344528Z caller=intrumentation.go:75 level=info msg="changing probe status" status=healthy
ts=2023-11-02T20:52:37.001397229Z caller=http.go:73 level=info service=http/server component=store msg="listening for requests and metrics" address=0.0.0.0:10902
ts=2023-11-02T20:52:37.001640506Z caller=tls_config.go:274 level=info service=http/server component=store msg="Listening on" address=:10902
ts=2023-11-02T20:52:37.001701496Z caller=tls_config.go:277 level=info service=http/server component=store msg="TLS is disabled." http2=false address=:10902
ts=2023-11-02T20:53:07.007659115Z caller=intrumentation.go:67 level=warn msg="changing probe status" status=not-ready reason="bucket store initial sync: sync block: BaseFetcher: iter bucket: Get \"http://thanos-inst-minio.thanos.svc.cluster.local:9000/thanos/?location=\": dial tcp 10.254.56.164:9000: connect: connection refused"
ts=2023-11-02T20:53:07.007717509Z caller=http.go:91 level=info service=http/server component=store msg="internal server is shutting down" err="bucket store initial sync: sync block: BaseFetcher: iter bucket: Get \"http://thanos-inst-minio.thanos.svc.cluster.local:9000/thanos/?location=\": dial tcp 10.254.56.164:9000: connect: connection refused"
ts=2023-11-02T20:53:07.007806312Z caller=intrumentation.go:56 level=info msg="changing probe status" status=ready
ts=2023-11-02T20:53:07.007845938Z caller=grpc.go:131 level=info service=gRPC/server component=store msg="listening for serving gRPC" address=0.0.0.0:10901
ts=2023-11-02T20:53:07.007933997Z caller=http.go:110 level=info service=http/server component=store msg="internal server is shutdown gracefully" err="bucket store initial sync: sync block: BaseFetcher: iter bucket: Get \"http://thanos-inst-minio.thanos.svc.cluster.local:9000/thanos/?location=\": dial tcp 10.254.56.164:9000: connect: connection refused"
ts=2023-11-02T20:53:07.007964385Z caller=intrumentation.go:81 level=info msg="changing probe status" status=not-healthy reason="bucket store initial sync: sync block: BaseFetcher: iter bucket: Get \"http://thanos-inst-minio.thanos.svc.cluster.local:9000/thanos/?location=\": dial tcp 10.254.56.164:9000: connect: connection refused"
ts=2023-11-02T20:53:07.008012055Z caller=intrumentation.go:67 level=warn msg="changing probe status" status=not-ready reason="bucket store initial sync: sync block: BaseFetcher: iter bucket: Get \"http://thanos-inst-minio.thanos.svc.cluster.local:9000/thanos/?location=\": dial tcp 10.254.56.164:9000: connect: connection refused"
ts=2023-11-02T20:53:07.008031272Z caller=grpc.go:138 level=info service=gRPC/server component=store msg="internal server is shutting down" err="bucket store initial sync: sync block: BaseFetcher: iter bucket: Get \"http://thanos-inst-minio.thanos.svc.cluster.local:9000/thanos/?location=\": dial tcp 10.254.56.164:9000: connect: connection refused"
ts=2023-11-02T20:53:07.00805076Z caller=grpc.go:151 level=info service=gRPC/server component=store msg="gracefully stopping internal server"
ts=2023-11-02T20:53:07.00808318Z caller=grpc.go:164 level=info service=gRPC/server component=store msg="internal server is shutdown gracefully" err="bucket store initial sync: sync block: BaseFetcher: iter bucket: Get \"http://thanos-inst-minio.thanos.svc.cluster.local:9000/thanos/?location=\": dial tcp 10.254.56.164:9000: connect: connection refused"
ts=2023-11-02T20:53:07.008216672Z caller=main.go:161 level=error err="Get \"http://thanos-inst-minio.thanos.svc.cluster.local:9000/thanos/?location=\": dial tcp 10.254.56.164:9000: connect: connection refused\nBaseFetcher: iter bucket\ngithub.com/thanos-io/thanos/pkg/block.(*BaseFetcher).fetchMetadata\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/block/fetcher.go:383\ngithub.com/thanos-io/thanos/pkg/block.(*BaseFetcher).fetch.func2\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/block/fetcher.go:456\ngithub.com/golang/groupcache/singleflight.(*Group).Do\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/pkg/mod/github.com/golang/groupcache@v0.0.0-20210331224755-41bb18bfe9da/singleflight/singleflight.go:56\ngithub.com/thanos-io/thanos/pkg/block.(*BaseFetcher).fetch\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/block/fetcher.go:454\ngithub.com/thanos-io/thanos/pkg/block.(*MetaFetcher).Fetch\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/block/fetcher.go:514\ngithub.com/thanos-io/thanos/pkg/store.(*BucketStore).SyncBlocks\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/store/bucket.go:556\ngithub.com/thanos-io/thanos/pkg/store.(*BucketStore).InitialSync\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/store/bucket.go:625\nmain.runStore.func5.1\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/cmd/thanos/store.go:427\ngithub.com/thanos-io/thanos/pkg/runutil.RetryWithLog\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/runutil/runutil.go:97\ngithub.com/thanos-io/thanos/pkg/runutil.Retry\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/runutil/runutil.go:87\nmain.runStore.func5\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/cmd/thanos/store.go:426\ngithub.com/oklog/run.(*Group).Run.func1\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38\nruntime.goexit\n\t/opt/bitnami/go/src/runtime/asm_amd64.s:1598\nsync block\ngithub.com/thanos-io/thanos/pkg/store.(*BucketStore).InitialSync\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/store/bucket.go:626\nmain.runStore.func5.1\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/cmd/thanos/store.go:427\ngithub.com/thanos-io/thanos/pkg/runutil.RetryWithLog\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/runutil/runutil.go:97\ngithub.com/thanos-io/thanos/pkg/runutil.Retry\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/pkg/runutil/runutil.go:87\nmain.runStore.func5\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/cmd/thanos/store.go:426\ngithub.com/oklog/run.(*Group).Run.func1\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38\nruntime.goexit\n\t/opt/bitnami/go/src/runtime/asm_amd64.s:1598\nbucket store initial sync\nmain.runStore.func5\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/cmd/thanos/store.go:432\ngithub.com/oklog/run.(*Group).Run.func1\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38\nruntime.goexit\n\t/opt/bitnami/go/src/runtime/asm_amd64.s:1598\nstore command failed\nmain.main\n\t/bitnami/blacksmith-sandox/thanos-0.32.5/src/github.com/thanos-io/thanos/cmd/thanos/main.go:161\nruntime.main\n\t/opt/bitnami/go/src/runtime/proc.go:250\nruntime.goexit\n\t/opt/bitnami/go/src/runtime/asm_amd64.s:1598"
Stream closed EOF for thanos/thanos-inst-storegateway-0 (storegateway)
```

### Additional information

I run on a Kubernetes 1.28.3 based on OpenStack.
My standard storage provisioner is cinder.csi.openstack.org

**Comment by carrodher** [comment]
Bitnami containers are designed to operate as non-root by default. Consequently, any files or directories used by the application should be owned by the _root_ group, as the random user (_1001_ by default) is a member of this _root_ group. To ensure proper permissions, you'll need to adjust the ownership of your local directory accordingly.

For more comprehensive information about non-root containers and their significance for security, you can explore the following resources:
* [Why non-root containers are important for security](https://docs.bitnami.com/tutorials/why-non-root-containers-are-important-for-security)
* [Use Non-Root Containers](https://docs.bitnami.com/kubernetes/faq/configuration/use-non-root/)
* [Work With Non-Root Containers for Bitnami Applications](https://docs.bitnami.com/tutorials/work-with-non-root-containers)
* [Running non-root containers on Openshift](https://docs.bitnami.com/tutorials/running-non-root-containers-on-openshift)

These references provide valuable insights into the best practices and considerations when working with non-root containers in Bitnami applications.

**Comment by lony** [comment]
Hello @carrodher thanks for your answer, sadly I do not get it 100 %.

Do you mean I should disable root access for the security context of a container? If so, which one do you mean?

I follow the advice from your chart documentation [here](https://artifacthub.io/packages/helm/bitnami/thanos#integrate-thanos-with-prometheus-and-alertmanager), which does not seem to change the security context. Only the `rootUser` and `rootPassword` for minio is changed, but I assumed this is the "access user" and has nothing to do with the security context.

I also run the charts as given and do not change the security context from outside. As I got your linked guides, I could add a security context, but shouldn't this already be set properly by the charts?

Can you please help me and explain what you meant?

**Comment by joancafom** [comment]
Hi @lony 

I have tried to reproduced your issue locally (I am using Minikube), but unfortunately cannot reproduce it. Everything seems to be deployed perfectly and with ease:

```console
$ k get pods -n thanos
NAME                                                          READY   STATUS    RESTARTS   AGE
alertmanager-thanos-inst-kube-prometheu-alertmanager-0        2/2     Running   0          16m
prometheus-thanos-inst-kube-prometheu-prometheus-0            2/2     Running   0          13m
thanos-inst-bucketweb-74cc9cfcdc-ngkcn                        1/1     Running   0          13m
thanos-inst-compactor-6545bf488c-j7n8n                        1/1     Running   0          13m
thanos-inst-kube-prometheu-blackbox-exporter-9844dd45-hzwbk   1/1     Running   0          16m
thanos-inst-kube-prometheu-operator-5d75fd8fd-8cc2c           1/1     Running   0          16m
thanos-inst-kube-state-metrics-5f9b496f8d-4n8st               1/1     Running   0          16m
thanos-inst-minio-76b949f488-m8kgn                            1/1     Running   0          13m
thanos-inst-node-exporter-zgxwr                               1/1     Running   0          16m
thanos-inst-query-79c895d9f4-qkzrc                            1/1     Running   0          13m
thanos-inst-query-frontend-6c94fd66df-g8l22                   1/1     Running   0          13m
thanos-inst-ruler-0                                           1/1     Running   0          13m
thanos-inst-storegateway-0                                    1/1     Running   0          13m
```

Having said that and taking a look at the trace you shared from minio, it all looks like a permissions issue with the FileSystem (as my colleague said)

```console
...
 20:53:03.97 
 20:53:03.97 Welcome to the Bitnami minio container
 20:53:03.97 Subscribe to project updates by watching https://github.com/bitnami/containers
 20:53:03.98 Submit issues and feature requests at https://github.com/bitnami/containers/issues
 20:53:03.98 
 20:53:03.98 INFO  ==> ** Starting MinIO setup **
/opt/bitnami/scripts/libminio.sh: line 364: /bitnami/minio/data/.root_user: Permission denied
Stream closed EOF for thanos/thanos-inst-minio-59785f96cd-4p8nv (minio)
```

The way file system behaves highly depends on the cluster (and provisioning type) provider. If for some reason, the permissions are not the expected ones (a random user -_1001_ by default-, member of the  _root_ group), the set up will fail. Could you please execute the following commands in order to obtain further info?

```console
$ kubectl describe pod/thanos-inst-minio-76b949f488-m8kgn -n thanos
$ kubectl exec pod/thanos-inst-minio-76b949f488-m8kgn -n thanos -- bash -c 'id; ls -la /bitnami/minio/; ls -la /bitnami/minio/data'
```

Please, bear in mind that you should change `pod/thanos-inst-minio-76b949f488-m8kgn` to the exact name of your `minio` pod to work. In addition, make sure the pod is in state `Running` when using the second command.

**Comment by lony** [comment]
Hello @joancafom,

thanks for your response. I also strongly assume this is a problem related to the storage of my cluster. I run on Openstack, and already had problems with other charts. Furthermore, I attached my storage and the storage class as well, maybe it helps.

I did test once again with the version from above as witht the latest versions (prometheus=8.22.5, thanos=12.16.1), but still have the problem.

The following outputs are from the "old" charts (prometheus=8.22.0, thanos=12.13.13).

_`kubectl describe pod/thanos-inst-minio-59785f96cd-42rgg -n thanos | pbcopy`_
```
Name:             thanos-inst-minio-59785f96cd-42rgg
Namespace:        thanos
Priority:         0
Service Account:  thanos-inst-minio
Node:             jj-test-iojiccjxlai6-node-2/10.0.0.239
Start Time:       Sun, 26 Nov 2023 17:37:37 +0100
Labels:           app.kubernetes.io/instance=thanos-inst
                  app.kubernetes.io/managed-by=Helm
                  app.kubernetes.io/name=minio
                  app.kubernetes.io/version=2023.10.16
                  helm.sh/chart=minio-12.8.15
                  pod-template-hash=59785f96cd
Annotations:      checksum/credentials-secret: 2b2ec4d8d178eabd1cd6d572e274efa5d6912b671d30fd03ebc24f59c436681a
Status:           Running
IP:               10.100.2.60
IPs:
  IP:           10.100.2.60
Controlled By:  ReplicaSet/thanos-inst-minio-59785f96cd
Containers:
  minio:
    Container ID:   containerd://de796c8bb8bb3c2c62a116565c4c973daeeaa9145338af3601aa8f15f414fb05
    Image:          docker.io/bitnami/minio:2023.10.16-debian-11-r0
    Image ID:       docker.io/bitnami/minio@sha256:3843dd641f455fdf6a7ec15adc3fb532c225a5cac13208e8f027c1bb0466f324
    Ports:          9000/TCP, 9001/TCP
    Host Ports:     0/TCP, 0/TCP
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Sun, 26 Nov 2023 17:39:19 +0100
      Finished:     Sun, 26 Nov 2023 17:39:20 +0100
    Ready:          False
    Restart Count:  4
    Liveness:       http-get http://:minio-api/minio/health/live delay=5s timeout=5s period=5s #success=1 #failure=5
    Readiness:      tcp-socket :minio-api delay=5s timeout=1s period=5s #success=1 #failure=5
    Environment:
      BITNAMI_DEBUG:               false
      MINIO_SCHEME:                http
      MINIO_FORCE_NEW_KEYS:        no
      MINIO_ROOT_USER:             <set to the key 'root-user' in secret 'thanos-inst-minio'>      Optional: false
      MINIO_ROOT_PASSWORD:         <set to the key 'root-password' in secret 'thanos-inst-minio'>  Optional: false
      MINIO_DEFAULT_BUCKETS:       thanos
      MINIO_BROWSER:               on
      MINIO_PROMETHEUS_AUTH_TYPE:  public
      MINIO_CONSOLE_PORT_NUMBER:   9001
    Mounts:
      /bitnami/minio/data from data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-b8mfz (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  thanos-inst-minio
    ReadOnly:   false
  kube-api-access-b8mfz:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason                  Age                  From                     Message
  ----     ------                  ----                 ----                     -------
  Normal   Scheduled               2m17s                default-scheduler        Successfully assigned thanos/thanos-inst-minio-59785f96cd-42rgg to jj-test-iojiccjxlai6-node-2
  Normal   SuccessfulAttachVolume  2m10s                attachdetach-controller  AttachVolume.Attach succeeded for volume "pvc-5582c18d-9541-45c8-ac4c-1f82d26895cd"
  Warning  BackOff                 46s (x10 over 2m4s)  kubelet                  Back-off restarting failed container minio in pod thanos-inst-minio-59785f96cd-42rgg_thanos(0b03db0c-463d-4915-a419-be1de57add5f)
  Normal   Pulled                  35s (x5 over 2m6s)   kubelet                  Container image "docker.io/bitnami/minio:2023.10.16-debian-11-r0" already present on machine
  Normal   Created                 35s (x5 over 2m6s)   kubelet                  Created container minio
  Normal   Started                 35s (x5 over 2m6s)   kubelet                  Started container minio
```

Sadly, my minio container never reaches `Running` state and is stuck in `CrashLoopBackOff`, therefore I can not execute the second command. The log of the container is the following

```
 16:40:43.98 
 16:40:43.98 Welcome to the Bitnami minio container
 16:40:43.98 Subscribe to project updates by watching https://github.com/bitnami/containers
 16:40:43.98 Submit issues and feature requests at https://github.com/bitnami/containers/issues
 16:40:43.98 
 16:40:43.98 INFO  ==> ** Starting MinIO setup **
/opt/bitnami/scripts/libminio.sh: line 364: /bitnami/minio/data/.root_user: Permission denied
Stream closed EOF for thanos/thanos-inst-minio-59785f96cd-42rgg (minio)
```

I hope this guides us further down the rabbit hole.

Thanks for your response and kind regards,
Goetz


**Attachment** 

_Storage_
```
Name:         cinder.csi.openstack.org
Namespace:    
Labels:       <none>
Annotations:  <none>
API Version:  storage.k8s.io/v1
Kind:         CSIDriver
Metadata:
  Creation Timestamp:  2021-09-07T14:32:32Z
  Resource Version:    482
  UID:                 7ee750fd-9e50-4861-878e-7692adf24a49
Spec:
  Attach Required:     true
  Fs Group Policy:     ReadWriteOnceWithFSType
  Pod Info On Mount:   true
  Requires Republish:  false
  Se Linux Mount:      false
  Storage Capacity:    false
  Volume Lifecycle Modes:
    Persistent
    Ephemeral
Events:  <none>
```

_StorageClass_
```
Name:            standard
IsDefaultClass:  Yes
Annotations:     kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"storage.k8s.io/v1","kind":"StorageClass","metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"true"},"name":"standard"},"provisioner":"cinder.csi.openstack.org","reclaimPolicy":"Delete","volumeBindingMode":"WaitForFirstConsumer"}
,storageclass.kubernetes.io/is-default-class=true
Provisioner:           cinder.csi.openstack.org
Parameters:            <none>
AllowVolumeExpansion:  <unset>
MountOptions:          <none>
ReclaimPolicy:         Delete
VolumeBindingMode:     WaitForFirstConsumer
Events:                <none>
```

**Comment by lony** [comment]
Thanks for your help. The output is now:

````
stream logs failed container "minio" in pod "thanos-inst-minio-6857f4cd56-rztsw" is waiting to start: ContainerCreating for thanos/thanos-inst-minio-6857f4cd56-rztsw (minio)
stream logs failed container "minio" in pod "thanos-inst-minio-6857f4cd56-rztsw" is waiting to start: ContainerCreating for thanos/thanos-inst-minio-6857f4cd56-rztsw (minio)
#### DEBUG ####
uid=1001 gid=0(root) groups=0(root),1001
total 4
drwxr-xr-x. 3 root root   18 Oct 16 07:23 .
drwxr-xr-x. 3 root root   19 Oct 16 07:23 ..
drwxr-xr-x. 3 root root 4096 Dec  1 10:55 data
total 20
drwxr-xr-x. 3 root root  4096 Dec  1 10:55 .
drwxr-xr-x. 3 root root    18 Oct 16 07:23 ..
drwx------. 2 root root 16384 Dec  1 10:55 lost+found
###############
 10:55:58.05 
 10:55:58.06 Welcome to the Bitnami minio container
 10:55:58.06 Subscribe to project updates by watching https://github.com/bitnami/containers
 10:55:58.06 Submit issues and feature requests at https://github.com/bitnami/containers/issues
 10:55:58.06 
 10:55:58.06 INFO  ==> ** Starting MinIO setup **
/opt/bitnami/scripts/libminio.sh: line 364: /bitnami/minio/data/.root_user: Permission denied
Stream closed EOF for thanos/thanos-inst-minio-6857f4cd56-rztsw (minio)
```

**Comment by joancafom** [comment]
Interesting! So, as you can see, the failure is legit and the permissions are set to `root:root`. The `volumePermissions` init container should have acted on your installation and fixed those permissions. I think I have found the issue, and it is trivial in fact, but I have initially mistakenly read the indentation in your values.yaml file. Use the following this time and let's see the logs now :) 

```diff
thanos:
  objstoreConfig: |-
    type: s3
    config:
      bucket: thanos
      endpoint: {{ include "thanos.minio.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local:9000
      access_key: minio
      secret_key: minio123
      insecure: true
  query:
    dnsDiscovery:
      sidecarsService: kube-prometheus-prometheus-thanos
      sidecarsNamespace: thanos
  bucketweb:
    enabled: true
  compactor:
    enabled: true
  storegateway:
    enabled: true
  ruler:
    enabled: true
    alertmanagers:
      - http://kube-prometheus-alertmanager.thanos.svc.cluster.local:9093
    config: |-
      groups:
        - name: "metamonitoring"
          rules:
            - alert: "PrometheusDown"
              expr: absent(up{prometheus="thanos/kube-prometheus"})
  metrics:
    enabled: true
    serviceMonitor:
      enabled: true
  minio:
    enabled: true
    command:
      - /bin/bash
    args:
      - -ec
      - |
        echo "#### DEBUG ####"
        id; ls -la /bitnami/minio/; ls -la /bitnami/minio/data
        echo "###############"
        /opt/bitnami/scripts/minio/entrypoint.sh /opt/bitnami/scripts/minio/run.sh
    auth:
      rootPassword: minio123
      rootUser: minio
    monitoringBuckets: thanos
    accessKey:
      password: minio
    secretKey:
      password: minio123
+   volumePermissions:
+     enabled: true
  volumePermissions:
    enabled: true
```

We were missing the `volumePermissions` entry at `minio` level, which I mistakenly read the first time with the one that is right below it (the one for `thanos`)

**Comment by lony** [comment]
Amazing! This solved it and It's now running.

THANKS A LOT! 🎆 🎉 

```
minio #### DEBUG ####
minio uid=1001 gid=0(root) groups=0(root),1001
minio total 4
minio drwxr-xr-x. 3 root root   18 Oct 16 07:23 .
minio drwxr-xr-x. 3 root root   19 Oct 16 07:23 ..
minio drwxr-xr-x. 3 1001 1001 4096 Dec  1 17:12 data
minio total 20
minio drwxr-xr-x. 3 1001 1001  4096 Dec  1 17:12 .
minio drwxr-xr-x. 3 root root    18 Oct 16 07:23 ..
minio drwx------. 2 1001 1001 16384 Dec  1 17:12 lost+found
minio ###############
minio  17:12:11.23 
minio  17:12:11.24 Welcome to the Bitnami minio container
minio  17:12:11.24 Subscribe to project updates by watching https://github.com/bitnami/containers
minio  17:12:11.24 Submit issues and feature requests at https://github.com/bitnami/containers/issues
minio  17:12:11.24 
minio  17:12:11.24 INFO  ==> ** Starting MinIO setup **
minio minio 17:12:11.33 INFO  ==> Starting MinIO in background...
minio minio 17:12:16.53 INFO  ==> Adding local Minio host to 'mc' configuration...
minio minio 17:12:16.76 INFO  ==> Creating default buckets...
minio Bucket created successfully `local/thanos`.
minio minio 17:12:17.06 INFO  ==> Stopping MinIO...
minio 
minio  17:12:17.15 INFO  ==> ** MinIO setup finished! **
minio minio 17:12:17.17 INFO  ==> ** Starting MinIO **
minio MinIO Object Storage Server
minio Copyright: 2015-2023 MinIO, Inc.
minio License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
minio Version: DEVELOPMENT.2023-10-16T04-13-43Z (go1.20.10 linux/amd64)
minio 
minio Status:         1 Online, 0 Offline. 
minio S3-API: http://localhost:9000 
minio Console: http://10.100.2.68:9001 http://127.0.0.1:9001   
minio 
minio Documentation: https://min.io/docs/minio/linux/index.html
minio Warning: The standard parity is set to 0. This can lead to data loss.
minio 
minio  You are running an older version of MinIO released 1 month before the latest release 
minio  Update: Run `mc admin update` 
minio 
minio 
Stream closed EOF for thanos/thanos-inst-minio-b6bf8b688-56mm2 (volume-permissions)
```
